### ISSUE TITLE
Incorrect display of nested maps

### ISSUE THREAD

**Issue by stepankoci** [getsentry/sentry-clj#36]
### Environment

How do you use Sentry?
- Sentry SaaS (sentry.io)

Which SDK and version?
- `io.sentry/sentry-clj {:mvn/version "5.6.169"}`

### Steps to Reproduce

When sending event extra data, they are not displayed correctly in the issue detail.

```clojure
(sentry/init! (:sentry/dsn "..."))
  
(sentry/send-event
 {:message "Test"
  :extra {:var1 "val1"
          :var2 {:a {:b [:c :d {:e :f} #{:g :h}]}}}})

(sentry/send-event
 {:throwable (ex-info "Sample exception"
                      {:var1 "val1"
                       :var2 {:a {:b [:c :d {:e :f} #{:g :h}]}}})}))
```

### Expected Result

Correct display of nested data structures in issue detail.

### Actual Result

Formatted view
<img width="839" alt="Screenshot 2022-02-11 v 15 23 00" src="https://user-images.githubusercontent.com/566808/153608874-4ffc7b4b-872e-4470-bd91-df260c56ae24.png">

Raw view:
```json
{"a":{"b":[{"hasheq":-1763192079,"sym":{"_hasheq":-122660552,"name":"c"}},{"hasheq":1972142424,"sym":{"_hasheq":-682293345,"name":"d"}},{"e":{"hasheq":-1597136552,"sym":"{\"_hasheq\":43394975,\"name\":\"f\"}"}},[{"hasheq":1738089905,"sym":"{\"_hasheq\":-916345864,\"name\":\"g\"}"},{"hasheq":1109658740,"sym":"{\"_hasheq\":-1544777029,\"name\":\"h\"}"}]]}}
```

**Comment by bruno-garcia** [comment]
I'm not sure if this is an SDK issue or Sentry's rendering. @dharrigan maybe has thoughts?

**Comment by dharrigan** [comment]
Hi,

Thank you for your report.

The issue here is that the `setExtra` method only permits key/values of Strings (as detailed here https://github.com/getsentry/sentry-java/blob/43622295a0c0a264113017f8c4968dbeede637ff/sentry/src/main/java/io/sentry/Sentry.java#L427).

Thus, support for vectors or sets is not supported as values. 

There is a bug in that the value isn't being converted in some cases to a String. That, I have addressed and will push out a bugfix release soon.

If I may, I've noticed that `setExtra` is been deprecated in favour of `setContext` (as noted here: https://docs.sentry.io/platforms/java/enriching-events/context/#additional-data).

After my bugfix, and in keeping with the values being either a String, a keyword or another map, if I have this:

`(send-event {:message "Test" :extra {:var1 "val1" :var2 {:a {:b {:c {:d {:e :f} :g :h}}}}}})`

I end up with this on Sentry:

![sentry](https://user-images.githubusercontent.com/427150/155843021-e5d0e613-f7ef-4ff1-a045-feae7eca72b8.png)

I hope that is of some help and thank you for raising this, expect a new release soon to ensure that each (keyword) value of the map is converted correctly to a string.
